scripts/dockerbot/check_prs.py

```diff
@@ -277,9 +313,19 @@ def check_status(pr):
                 logger.info('Setting task {} for pr {} to {}'
                             .format(task['stream'], pr['number'], states[task['result']]))
                 set_status(commit, states[task['result']], task['stream'])
+            task_states[task['stream']] = states[task['result']]
```

Comment:
>it will still append the "failure" result and print it out each time

The results will be printed only after all the remaining tasks will change their state from 'pending'/'running' to 'passed' or 'failed'. After one of the tasks has failed we won't receive a notification due to checks in line 320.

After all tasks have changed to 'passed'/'failed' notification won't be duplicated as ```check_status``` is triggered for pending objects only (see line 154)

I guess we'll miss a notification for a new run here (when tasks change to 'invalid'), but that's probably not what we're interested in anyway.